File: php/features.php
<?php
declare(strict_types=1);

// Irail SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class IrailFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new IrailBaseFeature();
            case "test":
                return new IrailTestFeature();
            default:
                return new IrailBaseFeature();
        }
    }
}

File: py/features.py
# Irail SDK feature factory

from feature.base_feature import IrailBaseFeature
from feature.test_feature import IrailTestFeature


def _make_feature(name):
    features = {
        "base": lambda: IrailBaseFeature(),
        "test": lambda: IrailTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
